app/controllers/stages_controller.rb

```diff
@@ -102,4 +99,12 @@ def stage_params
   def find_stage
     @stage = current_project.stages.find_by_param!(params[:id])
   end
+
+  def template_param
+    @template_param ||= ActiveRecord::Type::Boolean.new.type_cast_from_database(stage_params.delete("template"))
+  end
+
+  def update_template
+    template_param ? @stage.project.template!(@stage) : @stage.project.template!
```

Comment:
Also: this can't be an after_save hook because the default behavior will still set the passed in stage as template w/o unsetting the previous stage.